### DISCUSSION TITLE
feature: Use a Set to check for uniqueness in Array

### DISCUSSION THREAD

**Discussion by aarontravass** [typestack/class-validator#2367] [Feature Ideas]
### Description
The current `@ArrayUnique` uses a filter based method to compare objects. This is an `O(n^2)` solutions. The [JS spec for Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#description) states that

> The specification requires sets to be implemented "that, on average, provide access times that are sublinear on the number of elements in the collection". Therefore, it could be represented internally as a hash table (with O(1) lookup), a search tree (with O(log(N)) lookup), or any other data structure, as long as the complexity is better than O(N).

Hence, I believe it would be more performant to use a Set based approach to check for uniqueness. 

### Proposed solution
Something as simple as 

```ts
Array.from(new Set(inputArray)).length == inputArray.length
```

**Comment by braaar** [comment]
I'm moving this to the discussion tab, as we are trying to keep the issue count low. Please see [this issue](https://github.com/typestack/class-validator/issues/1775) for an explanation.
